drivers/spi/spi_mcux_flexio.c

```diff
@@ -0,0 +1,446 @@
+
+	FLEXIO_SetTimerConfig(base->flexioBase, base->timerIndex[0], &timerConfig);
+}
+
+static int spi_mcux_flexio_configure(const struct device *dev,
+	const struct spi_config *spi_cfg)
+{
+	const struct spi_mcux_flexio_config *config = dev->config;
+	struct spi_mcux_flexio_data *data = dev->data;
+
+	flexio_spi_master_config_t master_config;
+	uint32_t clock_freq;
+	uint32_t word_size;
+
+	if (spi_context_configured(&data->ctx, spi_cfg)) {
+		/* This configuration is already in use */
+		return 0;
+	}
+
+	if (spi_cfg->operation & SPI_HALF_DUPLEX) {
+		LOG_ERR("Half-duplex not supported");
+		return -ENOTSUP;
+	}
+
+	if (SPI_OP_MODE_GET(spi_cfg->operation) != SPI_OP_MODE_MASTER) {
+		LOG_ERR("Mode Slave not supported");
+		return -ENOTSUP;
+	}
+
+	FLEXIO_SPI_MasterGetDefaultConfig(&master_config);
+
+	word_size = SPI_WORD_SIZE_GET(spi_cfg->operation);
+	if ((word_size != 8) && (word_size != 16) && (word_size != 32)) {
+		LOG_ERR("Word size %d must be 8, 16 or 32", word_size);
+		return -EINVAL;
+	}
+	master_config.dataMode = word_size;
+
+	if (spi_cfg->operation & SPI_TRANSFER_LSB) {
+		if (word_size == 8) {
+			data->transfer_flags = kFLEXIO_SPI_8bitLsb;
+		} else if (word_size == 16) {
+			data->transfer_flags = kFLEXIO_SPI_16bitLsb;
+		} else {
+			data->transfer_flags = kFLEXIO_SPI_32bitLsb;
+		}
+	} else {
+		if (word_size == 8) {
+			data->transfer_flags = kFLEXIO_SPI_8bitMsb;
+		} else if (word_size == 16) {
+			data->transfer_flags = kFLEXIO_SPI_16bitMsb;
+		} else {
+			data->transfer_flags = kFLEXIO_SPI_32bitMsb;
+		}
+	}
+
+	if (mcux_flexio_get_rate(config->flexio_dev, &clock_freq)) {
+		return -EINVAL;
+	}
+
+	master_config.phase =
+		(SPI_MODE_GET(spi_cfg->operation) & SPI_MODE_CPHA)
+		? kFLEXIO_SPI_ClockPhaseSecondEdge
+		: kFLEXIO_SPI_ClockPhaseFirstEdge;
+
+	master_config.baudRate_Bps = spi_cfg->frequency;
+	spi_flexio_master_init(config->flexio_spi, &master_config,
+		(SPI_MODE_GET(spi_cfg->operation) & SPI_MODE_CPOL), clock_freq);
+
+	FLEXIO_SPI_MasterTransferCreateHandle(config->flexio_spi, &data->handle,
+					spi_mcux_master_transfer_callback,
+					data);
+	/* No SetDummyData() for FlexIO_SPI */
+
+	data->ctx.config = spi_cfg;
+
+	return 0;
+}
+
+
+static int transceive(const struct device *dev,
+			const struct spi_config *spi_cfg,
+			const struct spi_buf_set *tx_bufs,
+			const struct spi_buf_set *rx_bufs,
+			bool asynchronous,
+			spi_callback_t cb,
+			void *userdata)
+{
+	const struct spi_mcux_flexio_config *config = dev->config;
+	struct spi_mcux_flexio_data *data = dev->data;
+	int ret;
+
+	spi_context_lock(&data->ctx, asynchronous, cb, userdata, spi_cfg);
+
+	mcux_flexio_lock(config->flexio_dev);
+	ret = spi_mcux_flexio_configure(dev, spi_cfg);
+	mcux_flexio_unlock(config->flexio_dev);
+	if (ret) {
+		goto out;
+	}
+
+	spi_context_buffers_setup(&data->ctx, tx_bufs, rx_bufs, 1);
+
+	spi_context_cs_control(&data->ctx, true);
+
+	mcux_flexio_lock(config->flexio_dev);
+	mcux_flexio_irq_disable(config->flexio_dev);
```

Comment:
I'm unclear here- why do we need to disable the IRQ from the FlexIO before sending the first packet? I'm concerned when multiple FlexIO "devices" are defined, that disabling the interrupt could have adverse affects

Reply:
@danieldegrasse, Because modification of general data occurs there: https://github.com/zephyrproject-rtos/zephyr/pull/53748/files#diff-d2bd4d8ff2692193e151b71d1ff172bfeca0ca52c197506a8c8548cabf05ff44R94